subsys/net/lib/sockets/sockets.c

```diff
@@ -1343,6 +1343,10 @@ static inline ssize_t zsock_recv_dgram(struct net_context *ctx,
 			 * point. The best we can do is returning remote address
 			 * if that was set using connect() call.
 			 */
+
+            (void)net_context_recvfrom(ctx, zsock_received_cb, K_NO_WAIT,
```

Comment:
I kind of miss the idea here, we've already received packet at this point from the fifo, what's the point of re-registering the recv callback function now?